nav2_rviz_plugins/src/nav2_panel.cpp

```diff
@@ -48,29 +58,99 @@ Nav2Panel::Nav2Panel(QWidget * parent)
+
+  // State entered after the NavigateToPose action has completed
+  completed_ = new QState();
+  completed_->setObjectName("succesful");
+  completed_->assignProperty(start_stop_button_, "text", "Shutdown");
+  completed_->assignProperty(start_stop_button_, "toolTip", shutdown_msg);
+
+  // State entered while the NavigateToPose action is active
+  running_ = new QState();
+  running_->setObjectName("running");
+  running_->assignProperty(start_stop_button_, "text", "Cancel");
+  running_->assignProperty(start_stop_button_, "toolTip", cancel_msg);
+
   stopping_ = new QState();
   stopping_->setObjectName("stopping");
   stopping_->assignProperty(start_stop_button_, "enabled", false);
 
   QObject::connect(starting_, SIGNAL(entered()), this, SLOT(onStartup()));
   QObject::connect(stopping_, SIGNAL(entered()), this, SLOT(onShutdown()));
+  QObject::connect(canceled_, SIGNAL(entered()), this, SLOT(onCancel()));
 
   initial_->addTransition(start_stop_button_, SIGNAL(clicked()), starting_);
   starting_->addTransition(start_stop_button_, SIGNAL(clicked()), stopping_);
+  running_->addTransition(start_stop_button_, SIGNAL(clicked()), canceled_);
+  canceled_->addTransition(start_stop_button_, SIGNAL(clicked()), stopping_);
+  completed_->addTransition(start_stop_button_, SIGNAL(clicked()), stopping_);
 
-  machine_.addState(initial_);
-  machine_.addState(starting_);
-  machine_.addState(stopping_);
+  ROSActionQTransition * startupTransition = new ROSActionQTransition(false);
+  startupTransition->setTargetState(running_);
+  starting_->addTransition(startupTransition);
 
-  machine_.setInitialState(initial_);
-  machine_.start();
+  ROSActionQTransition * canceledTransition = new ROSActionQTransition(false);
+  canceledTransition->setTargetState(running_);
+  canceled_->addTransition(canceledTransition);
 
-  // Lay out the items in the panel
+  ROSActionQTransition * runningTransition = new ROSActionQTransition(true);
+  runningTransition->setTargetState(completed_);
+  running_->addTransition(runningTransition);
 
+  ROSActionQTransition * completedTransition = new ROSActionQTransition(false);
+  completedTransition->setTargetState(running_);
+  completed_->addTransition(completedTransition);
+
+  state_machine_.addState(initial_);
+  state_machine_.addState(starting_);
+  state_machine_.addState(stopping_);
+  state_machine_.addState(running_);
+  state_machine_.addState(canceled_);
+  state_machine_.addState(completed_);
+
+  state_machine_.setInitialState(initial_);
+  state_machine_.start();
+
+  // Lay out the items in the panel
   QVBoxLayout * main_layout = new QVBoxLayout;
   main_layout->addWidget(start_stop_button_);
   main_layout->setContentsMargins(10, 10, 10, 10);
   setLayout(main_layout);
+
+  auto options = rclcpp::NodeOptions().arguments(
+    {"__node:=navigation_dialog_action_client"});
+  client_node_ = std::make_shared<rclcpp::Node>("_", options);
```

Comment:
What are you doing here? Why create the node by the name of "_" but then use a argument to rename it?

Reply:
This wasn't my code but moved from the `navigation_dialog.cpp` before my refactor. Seemingly, this is done several places in the code (see `bt_navigator.cpp`). If I recall, this was done to support remapping?